src/sentry/api/endpoints/group_similar_issues.py

```diff
@@ -38,22 +38,33 @@ def get(self, request, group):
         if limit is not None:
             limit = int(limit) + 1  # the target group will always be included
 
-        raw_results = {
-            group_id: {_fix_label(label): features for label, features in scores.items()}
-            for group_id, scores in features.compare(group, limit=limit)
-            if group_id != group.id
+        group_ids = []
+        group_scores = []
+
+        for group_id, scores in features.compare(group, limit=limit):
+            if group_id != group.id:
+                group_ids.append(group_id)
+                group_scores.append(scores)
+
+        serialized_groups = {
+            g["id"]: g
+            for g in serialize(
+                list(Group.objects.get_many_from_cache(group_ids)), user=request.user
+            )
         }
 
-        fetched_groups = Group.objects.get_many_from_cache(list(raw_results))
-        serialized_groups = serialize(fetched_groups, user=request.user)
-        group_scores = list(raw_results.pop(group.id) for group in fetched_groups)
+        results = []
 
-        results = list(zip(serialized_groups, group_scores))
+        # We need to preserve the ordering of the Redis results, as that
+        # ordering is directly shown in the UI
+        for group_id, scores in zip(group_ids, group_scores):
+            group = serialized_groups.get(group_id)
+            if group is None:
+                # TODO(tkaemming): This should log when we filter out a group that is
```

Comment:
@tkaemming I attempted to add logging here, but it actually happens too often to be worthwhile logging to Sentry. What are the reasons one would get invalid group ids back from similarity?

Reply:
This can happen any time an events relationship with a group is mutated (merging, unmerging, etc) either temporarily due to inconsistencies between different datastores in the case of in-flight updates that haven't been reflected yet somewhere, or less temporarily if there was a failure to perform the update that never got retried for some reason resulting in inconsistent state that lingers around until it expires.

Reply:
ok but this happens so often I feel like there must be something else at play, other than temporary inconsistencies while merging. I guess I'll just leave it like that then

Reply:
I wouldn't be too surprised if something is broken on the write path that leads to this getting inconsistent over time (I also think deletions might be a factor here thinking about it.) I don't think we have enough retention to see if this has gotten worse over time but I don't recall it being an issue when it was initially developed so we may have broken something in the meantime, I might be misremembering though.